src/gluonts/dataset/split.py

```diff
@@ -350,6 +354,10 @@ def test_pair(
         base = periods_between(entry["start"], self.date)
         input_slice = slice(None, base + offset)
         label_slice = slice(base + offset, base + offset + prediction_length)
+        assert (
+            base + offset + prediction_length
+            <= entry[FieldName.TARGET].shape[-1]
+        ), "Offset too short to generate windows"
```

Comment:
rhs of the inequality is just `label_slice.stop`, you could use that I guess. Also maybe in this case the message should be `"Date is too early to generate windows"`?

Reply:
Done

Reply:
My bad: `"Date is too late to generate windows"` :-)